### PR TITLE
[DLLM] Add initial cuda graph support

### PR DESCRIPTION
**Pull request by btw616**
## Motivation

Add cuda graph support for diffusion LLMs.

## Modifications

- Add a `DLLM_EXTEND` forward mode for dllm.
- Add cuda graph support for the `DLLM_EXTEND` forward mode.
- Extend flashinfer backend to support cuda graph for the `DLLM_EXTEND` forward mode.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.ai/developer_guide/contribution_guide.html#code-style-guidance).
- [ ] Work with maintainers to merge your PR. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process)

### CODE REVIEW

File: python/sglang/srt/server_args.py [hunk 1]
```diff
@@ -1995,10 +1995,16 @@ def _handle_dllm_inference(self):
...
+            if self.cuda_graph_bs != [1]:
+                logger.warning(
+                    "Cuda graph bs is set to [1] because of using diffusion LLM inference"
+                )
+                self.cuda_graph_bs = [1]
+            if self.attention_backend != "flashinfer":
+                logger.warning(
+                    "Attention backend is set to flashinfer because of enabling cuda graph in diffusion LLM inference"
+                )
+                self.attention_backend = "flashinfer"
```
[the remark this answers is not in the document]
Comment (btw616) [reply python/sglang/srt/server_args.py:2026]: Once dllm supports dynamic batching (roadmap: #14199), the `cuda_graph_bs=[1]` restriction can be removed.

File: python/sglang/srt/layers/attention/flashinfer_backend.py [hunk 2]
```diff
@@ -127,8 +128,8 @@ def __init__(
         )
 
         # FIXME: remove dllm workarounds from flashinfer
-        self.is_dllm_model = model_runner.server_args.dllm_algorithm is not None
-        self.dllm_block_size = model_runner.server_args.dllm_block_size
+        self.dllm_config = DllmConfig.from_server_args(model_runner.server_args)
```
Comment (Monstertail) [inline python/sglang/srt/layers/attention/flashinfer_backend.py:131]: It's great to move the block_size into the dllm_config.

Shall we support the parsing in the algorithm interface by the way? I can do that. @btw616

Comment (Monstertail) [reply python/sglang/srt/layers/attention/flashinfer_backend.py:131]: Mark it resolved, as dllm_config as server args will be supported in another PR.

File: python/sglang/srt/model_executor/cuda_graph_runner.py [hunk 3]
```diff
@@ -829,7 +838,14 @@ def replay(
         output = self.output_buffers[graph_key]
         if isinstance(output, LogitsProcessorOutput):
             return LogitsProcessorOutput(
-                next_token_logits=output.next_token_logits[: self.raw_num_token],
+                next_token_logits=(
```
Comment (hnyls2002) [inline python/sglang/srt/model_executor/cuda_graph_runner.py:841]: nits: do not put this logic inside `LogitsProcessorOutput`'s init method. Get a variable dispatch under this if branch.

**Review by hnyls2002 — APPROVED** [review]

**Review by BBuf — APPROVED** [review]
